src/index.js

```diff
@@ -3306,47 +3310,45 @@ async function retrieveMetadata() {
 
 	if ( queueItem ) {
 
-		let uri    = queueItem.dataset.file,
-			revoke = false;
+		let uri    = queueItem.dataset.file;
+		let file;
 
 		waitingMetadata++;
 		delete queueItem.dataset.retrieve;
+		let metadata;
 
-		queryMetadata: {
-			if ( queueItem.handle ) {
-				try {
-					if ( await queueItem.handle.requestPermission() != 'granted' )
-						break queryMetadata;
+		if ( queueItem.handle ) {
+			// Fetch metadata from File object
+			if ( await queueItem.handle.requestPermission() !== 'granted' )
+				return;
 
-					uri = URL.createObjectURL( await queueItem.handle.getFile() );
-					revoke = true;
-				}
-				catch( e ) {
-					break queryMetadata;
-				}
+			file = await queueItem.handle.getFile();
+			uri = URL.createObjectURL( file );
+			metadata = await parseBlob( file );
+		} else {
+			// Fetch metadata from URI
+			const response = await fetch(uri);
+			if (response.body && typeof response.body.getReader === 'function') {
+				metadata = await parseWebStream( response.body, { skipPostHeaders: true } );
+			} else {
+				// Fallback to Blob, in case the HTTP Result cannot be streamed
+				metadata = await parseBlob( await response.blob());
 			}
+		}
 
-			try {
-				const metadata = await mm.fetchFromUrl( uri, { skipPostHeaders: true } );
-				if ( metadata ) {
-					addMetadata( metadata, queueItem ); // add metadata to play queue item
-					syncMetadataToAudioElements( queueItem );
-					if ( ! ( metadata.common.picture && metadata.common.picture.length ) ) {
-						getFolderCover( queueItem ).then( cover => {
-							queueItem.dataset.cover = cover;
-							syncMetadataToAudioElements( queueItem );
-						});
-					}
-				}
-			}
-			catch( e ) {}
+		addMetadata( metadata, queueItem ); // add metadata to play queue item
+		syncMetadataToAudioElements( queueItem );
+		if ( ! queueItem.handle && ! ( metadata.common.picture && metadata.common.picture.length ) ) {
+			queueItem.dataset.cover = await getFolderCover( uri );
+			syncMetadataToAudioElements( queueItem );
```

Comment:
This seems to be old code from the master branch as well. `getFolderCover()` now takes the `queueItem` and works for both File object and URL, so the `! queueItem.handle` check is not wanted here anymore. I think this will do it:

```
		if ( ! ( metadata.common.picture && metadata.common.picture.length ) ) {
			queueItem.dataset.cover = await getFolderCover( queueItem );
			syncMetadataToAudioElements( queueItem );
		}
```

Reply:
The double check avoids an error in case `metadata.common.picture` is undefined. Could use `metadata.common.picture?.length`, but trying to keep compatible with older browsers..

Reply:
Doesn't really matter, result is the same. The undefined check is still there.

Reply:
> Doesn't really matter, result is the same. The undefined check is still there.

It's there, but it never executes.

<img width="271" height="152" alt="image" src="https://github.com/user-attachments/assets/38990bca-5238-447b-865e-21a7f6b45d89" />

Reply:
Loads `cover.jpg` from my folder just fine.
Maybe spin up with `npm run dev`, so you have actual line number?

Reply:
> Loads `cover.jpg` from my folder just fine. Maybe spin up with `npm run dev`, so you have actual line number?

Not an actual error from the app. I was just making a point why we need the `( metadata.common.picture && metadata.common.picture.length )` check. Maybe I misunderstood your previous comment.